### PR TITLE
Contains a workaround for the capitalized configuration issue

### PR DESCRIPTION
**Pull request by bastbnl**
This is a workaround for issue #6366. It doesn't emit the suggestion

### CODE REVIEW

File: celery/apps/worker.py [hunk 1]
```diff
@@ -141,12 +141,25 @@ def on_start(self):
...
-                "Please run `celery upgrade settings path/to/settings.py` "
-                "to avoid these warnings and to allow a smoother upgrade "
-                "to Celery 6.0."
-            )
+        # Ok this qualifies as a hack for issue #6366. Tried making it less of
+        # a hack via app.__reduce_keys__(), but that may not work properly in
+        # all cases
+        warn_deprecated = True
+        config_source = app._config_source
+        if isinstance(config_source, (str, )):
```
Comment (thedrow) [inline celery/apps/worker.py:149]:
```suggestion
        if isinstance(config_source, str):
```
A tuple is only required if we have multiple types to test `config_source` against.

File: celery/apps/worker.py [hunk 2]
```diff
@@ -141,12 +141,25 @@ def on_start(self):
             app.log.redirect_stdouts(self.redirect_stdouts_level)
 
         # TODO: Remove the following code in Celery 6.0
-        if app.conf.maybe_warn_deprecated_settings():
-            logger.warning(
-                "Please run `celery upgrade settings path/to/settings.py` "
-                "to avoid these warnings and to allow a smoother upgrade "
-                "to Celery 6.0."
-            )
+        # Ok this qualifies as a hack for issue #6366. Tried making it less of
```
Comment (thedrow) [inline celery/apps/worker.py:144]:
```suggestion
        # This qualifies as a hack for issue #6366.
```

**Review by thedrow — CHANGES_REQUESTED** [review]
Just a few nitpicks.
Otherwise, LGTM.

**Review by auvipy — APPROVED** [review]

**Comment by thedrow** [comment]
@auvipy couldn't you see I reviewed the PR and had some suggestions?
Why did you merge this anyway???

**Pull request merged**